File: static/js/atlasPersonality.js
// Atlas OS — dynamic assistant personality, addressing, and phrasing

const ASSISTANT_PROFILES = {
  Atlas: { id: 'Atlas', label: 'Atlas', tone: 'professional', defaultGender: 'male' },
  Atlasia: { id: 'Atlasia', label: 'Atlasia', tone: 'conversational', defaultGender: 'female' },
  Athena: { id: 'Athena', label: 'Athena', tone: 'professional', reserved: true },
  Oracle: { id: 'Oracle', label: 'Oracle', tone: 'executive', reserved: true },
  Sentinel: { id: 'Sentinel', label: 'Sentinel', tone: 'minimal', reserved: true },
  Nova: { id: 'Nova', label: 'Nova', tone: 'friendly', reserved: true },
};

function _settings() {
  return window.AtlasUserSettings?.getAtlasUserSettings?.() || {};
}

function _pick(list) {
  return list[Math.floor(Math.random() * list.length)] || '';
}

function _normalizeAddress(raw) {
  const val = String(raw || 'sir').trim().toLowerCase();
  if (val === 'maam' || val === 'madam') return "ma'am";
  if (val === 'none' || val === '') return '';
  if (val === 'sir' || val === 'boss' || val === "ma'am") return val;
  return 'sir';
}

export function getProfile() {
  const s = _settings();
  const identity = s.assistant_identity || 'Atlas';
  const base = { ...(ASSISTANT_PROFILES[identity] || ASSISTANT_PROFILES.Atlas) };
  return {
    ...base,
    identity,
    preferred_address: _normalizeAddress(s.preferred_address || s.address_style),
    response_style: s.response_style || 'professional',
    speech_rate: Number(s.speech_rate) || 1.0,
  };
}

export function getAddress() {
  return getProfile().preferred_address;
}

export function appendAddress(text) {
  const addr = getAddress();
  const clean = String(text || '').trim().replace(/[.!?]+$/, '');
  if (!clean) return '';
  if (!addr) return `${clean}.`;
  return `${clean}, ${addr}.`;
}

export function getGreeting() {
  const { identity, preferred_address: addr, response_style: style } = getProfile();
  if (identity === 'Atlasia') {
    const map = {
      professional: ['Online.', 'Ready when you are.', "I'm here."],
      friendly: ["Hey — I'm here.", 'Ready when you are.', 'Online.'],
      executive: ['Ready.', 'Online.', 'Standing by.'],
      minimal: ['Ready.', 'Online.'],
    };
    return _pick(map[style] || map.professional);
  }
  if (addr === 'boss') return _pick(['Online boss.', 'Yes boss.', 'Standing by boss.']);
  if (addr === "ma'am") return _pick(["Online ma'am.", "Yes ma'am.", "Standing by ma'am."]);
  if (addr === 'sir') return _pick(['Online sir.', 'Yes sir.', 'Standing by sir.']);
  return _pick(['Online.', 'Ready.', 'Standing by.']);
}

export function getStandby() {
  if (getProfile().identity === 'Atlasia') {
    return _pick(['Standing by.', "I'll be here.", 'On standby.']);
  }
  return appendAddress('Standing by');
}

export function getCompletion() {
  if (getProfile().identity === 'Atlasia') {
    return _pick(['All set.', 'Done.', 'Finished.', 'Everything is ready.', 'Research finished.']);
  }
  return appendAddress(_pick(['Done', 'Completed', 'Research complete']));
}

export function getConfirmation(action) {
  const text = String(action || '').trim().replace(/[.!?]+$/, '');
  if (!text) return getCompletion();
  if (getProfile().identity === 'Atlasia') return `${text}.`;
  return appendAddress(text);
}

export function getError(message) {
  const msg = String(message || 'something went wrong').trim().replace(/[.!?]+$/, '');
  if (getProfile().identity === 'Atlasia') {
    return msg.toLowerCase().startsWith('sorry') ? `${msg}.` : `Sorry — ${msg.toLowerCase()}.`;
  }
  if (!getAddress()) return `Sorry, ${msg.toLowerCase()}.`;
  return appendAddress(`Sorry, ${msg}`);
}

export function formatAction(action) {
  return getConfirmation(action);
}

export function formatQuestion(question) {
  const q = String(question || '').trim().replace(/\?$/, '');
  if (!q) return '';
  if (getProfile().identity === 'Atlasia') return `${q}?`;
  const addr = getAddress();
  if (!addr) return `${q}?`;
  return `${q}, ${addr}?`;
}

export const atlasPersonality = {
  ASSISTANT_PROFILES,
  getProfile,
  getAddress,
  appendAddress,
  getGreeting,
  getStandby,
  getCompletion,
  getConfirmation,
  getError,
  formatAction,
  formatQuestion,
};

window.AtlasProfile = atlasPersonality;
export default atlasPersonality;

File: src/atlas_personality.py
"""Atlas OS — dynamic assistant personality and addressing."""

from __future__ import annotations

import random
from typing import Any, Dict, List, Optional

from src.atlas_user_settings import load_user_settings

# Reserved for future assistants without backend churn.
ASSISTANT_PROFILES: Dict[str, Dict[str, Any]] = {
    "Atlas": {"id": "Atlas", "tone": "professional", "default_gender": "male"},
    "Atlasia": {"id": "Atlasia", "tone": "conversational", "default_gender": "female"},
    "Athena": {"id": "Athena", "tone": "professional", "reserved": True},
    "Oracle": {"id": "Oracle", "tone": "executive", "reserved": True},
    "Sentinel": {"id": "Sentinel", "tone": "minimal", "reserved": True},
    "Nova": {"id": "Nova", "tone": "friendly", "reserved": True},
}

_VALID_ADDRESS = {"sir", "boss", "ma'am", "maam", "none", ""}


def _pick(options: List[str]) -> str:
    return random.choice(options) if options else ""


def _normalize_address(raw: Optional[str]) -> str:
    val = (raw or "sir").strip().lower()
    if val in ("maam", "madam"):
        return "ma'am"
    if val in ("none", ""):
        return ""
    if val in ("sir", "boss", "ma'am"):
        return val
    return "sir"


def get_profile(settings: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
    s = settings or load_user_settings()
    identity = s.get("assistant_identity") or "Atlas"
    base = dict(ASSISTANT_PROFILES.get(identity, ASSISTANT_PROFILES["Atlas"]))
    base["identity"] = identity
    base["preferred_address"] = _normalize_address(
        s.get("preferred_address") or s.get("address_style")
    )
    base["response_style"] = s.get("response_style") or "professional"
    return base


def get_address(settings: Optional[Dict[str, Any]] = None) -> str:
    return get_profile(settings)["preferred_address"]


def _append_address(text: str, settings: Optional[Dict[str, Any]] = None) -> str:
    addr = get_address(settings)
    if not addr:
        return text if text.endswith((".", "!", "?")) else f"{text}."
    clean = text.rstrip(".!?")
    return f"{clean}, {addr}."


def get_greeting(settings: Optional[Dict[str, Any]] = None) -> str:
    profile = get_profile(settings)
    identity = profile["identity"]
    addr = profile["preferred_address"]
    style = profile["response_style"]

    if identity == "Atlasia":
        options = {
            "professional": ["Online.", "Ready when you are.", "I'm here."],
            "friendly": ["Hey — I'm here.", "Ready when you are.", "Online."],
            "executive": ["Ready.", "Online.", "Standing by."],
            "minimal": ["Ready.", "Online."],
        }
        return _pick(options.get(style, options["professional"]))

    if addr == "boss":
        return _pick(["Online boss.", "Yes boss.", "Standing by boss."])
    if addr == "ma'am":
        return _pick(["Online ma'am.", "Yes ma'am.", "Standing by ma'am."])
    if addr == "sir":
        return _pick(["Online sir.", "Yes sir.", "Standing by sir."])
    return _pick(["Online.", "Ready.", "Standing by."])


def get_standby(settings: Optional[Dict[str, Any]] = None) -> str:
    profile = get_profile(settings)
    if profile["identity"] == "Atlasia":
        return _pick(["Standing by.", "I'll be here.", "On standby."])
    return _append_address("Standing by", settings)


def get_completion(settings: Optional[Dict[str, Any]] = None) -> str:
    profile = get_profile(settings)
    if profile["identity"] == "Atlasia":
        return _pick(["All set.", "Done.", "Finished.", "Everything is ready."])
    return _append_address(_pick(["Done", "Completed", "Research complete"]), settings)


def get_confirmation(action: str, settings: Optional[Dict[str, Any]] = None) -> str:
    text = (action or "").strip().rstrip(".")
    if not text:
        return get_completion(settings)
    profile = get_profile(settings)
    if profile["identity"] == "Atlasia":
        return f"{text}."
    return _append_address(text, settings)


def get_error(message: str, settings: Optional[Dict[str, Any]] = None) -> str:
    msg = (message or "Something went wrong").strip().rstrip(".")
    profile = get_profile(settings)
    if profile["identity"] == "Atlasia":
        return f"Sorry — {msg.lower()}." if not msg.lower().startswith("sorry") else f"{msg}."
    if not get_address(settings):
        return f"Sorry, {msg.lower()}."
    return _append_address(f"Sorry, {msg}", settings)


def get_prompt_address(settings: Optional[Dict[str, Any]] = None) -> str:
    addr = get_address(settings)
    return addr or "the user respectfully"
